### PR TITLE
Fix diff output for data types where `-v` would show less information

### LINKED ISSUE #5192: Failed test output is less verbose with -v than not using -v
I don't know if this has changed recently but it seems in some cases using -v makes the output *less* verbose...

```
def test_verbosity():
    s = [{'SKU': '12345678', 'Name': 'Linda', 'Description': 'Fancy dress', 'Net Price': '9.16', 'VAT rate': '20', 'VAT': '1.83', 'Gross Price': '10.99', 'Colour': 'Grey', 'Image': 'FancyDressImage'}, {'SKU': '12345679', 'Name': 'Jane', 'Description': 'Skirt', 'Net Price': '10.00', 'VAT rate': '0', 'VAT': '0.00', 'Gross Price': '10.00', 'Colour': 'Pink', 'Image': 'ChildsSkirt'}]
    assert len(s) == 1
```

Normal output
```
$ pytest -k verbosity --tb=short
====================================== test session starts ======================================
platform linux -- Python 3.6.7, pytest-4.4.1, py-1.8.0, pluggy-0.9.0
rootdir: /mnt/c/Users/bjl/wsl/bdd-examples-project/tests, inifile: pytest.ini
plugins: bdd-3.1.0
collected 27 items / 26 deselected / 1 selected

steps/test_example.py F                                                                   [100%]

=========================================== FAILURES ============================================
________________________________________ test_verbosity _________________________________________
steps/test_example.py:55: in test_verbosity
    assert len(s) == 1
E   AssertionError: assert 2 == 1
E    +  where 2 = len([{'Colour': 'Grey', 'Description': 'Fancy dress', 'Gross Price': '10.99', 'Image': 'FancyDressImage', ...}, {'Colour': 'Pink', 'Description': 'Skirt', 'Gross Price': '10.00', 'Image': 'ChildsSkirt', ...}])
============================ 1 failed, 26 deselected in 0.12 seconds ============================
```

Output with -v
```
$ pytest -k verbosity --tb=short -v
====================================== test session starts ======================================
platform linux -- Python 3.6.7, pytest-4.4.1, py-1.8.0, pluggy-0.9.0 -- /home/bjl/.local/share/virtualenvs/bdd-examples-project-dSboamh_/bin/python3
cachedir: .pytest_cache
rootdir: /mnt/c/Users/bjl/wsl/bdd-examples-project/tests, inifile: pytest.ini
plugins: bdd-3.1.0
collected 27 items / 26 deselected / 1 selected

steps/test_example.py::test_verbosity FAILED                                              [100%]

=========================================== FAILURES ============================================
________________________________________ test_verbosity _________________________________________
steps/test_example.py:55: in test_verbosity
    assert len(s) == 1
E   assert 2 == 1
E     -2
E     +1
============================ 1 failed, 26 deselected in 0.12 seconds ============================
```

Calling it with higher levels of verbosity (-vv, -vvv) has the same output as -v.

Using pytest 4.4.1.

### PR DESCRIPTION
**Pull request by nicoddemus**
`_compare_eq_verbose` doesn't seem to useful in the end, with the examples we
have in the test suite.

Close #5192

cc @mdmintz

### CODE REVIEW

**Comment by mdmintz** [comment]
@nicoddemus Yes, this is much better. This resolves the issue seen earlier. I'm currently testing various assertion combinations, and so far this change seems to have the desired outcome.

**Review by bluetech — APPROVED** [review]
TL;DR - LGTM :)

IIUC the situation is this:

- Assertion rewrite calls the reprcompare mechanism, with custom "expls" for the assertion (the rewrite special sauce)
- But before using these special expls, it also calls the `pytest_assertrepr_compare` hook to see if it wants to offer its own formatting (which is preferred). The hook only gets the op and values (no special stringification sauce).
- The built-in `pytest_assertrepr_compare` hookimpl in this case eventually reaches `_compare_eq_any`.

Without verbose, `_compare_eq_any` returns None for this, and the custom expls are used.
With verbose, before this fix, `_compare_eq_verbose` would be called, which ends up with

1. Less nice formatting, because no special sauce.
2. Less verbose than the expls, because no special sauce, only shows the actual values.

This PR removes `_compare_eq_verbose` so it falls back to the rewrite expls.

So I looked into why `_compare_eq_any` was added in the first place. AFAIU, it used to be that the rewrite formatting didn't take verbosity in account when truncating - it always truncated. So `_compare_eq_any` was added as a workaround. But the underlying problem has since been fixed by @nicoddemus in be8d63e33b199cafa2656d49ad7128cc32214004. So this is probably no longer needed.

**Comment by nicoddemus** [comment]
Thanks @bluetech for looking into why `_compare_eq_verbose` existed in the first place. 😁 

Do you agree this is an improvement, or should we treat it as bug fix and backport to 7.0.x?

**Comment by bluetech** [comment]
I think the compare-any is an improvement, I wouldn't backport it since it's not risk-free either.

The `-q` is a bugfix, can backport if you feel like it :)

We probably shouldn't wait too long with pytest 7.1 though. I will open an issue for a release schedule.
